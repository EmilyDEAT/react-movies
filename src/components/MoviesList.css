.moviesListContainer {
  width: 90%;
  margin-left: auto;
  margin-right: auto;
  margin-top: 5vh;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-template-rows: auto;
  grid-gap: 2% 3% ;
}

.noFavorite {
  grid-column-start: 1;
  grid-column-end: 5;
}

@media screen and (min-width: 450px) and (max-width: 770px) {
  .moviesListContainer {
    grid-template-columns: repeat(4, 1fr);
  }
}

@media screen and (max-width: 450px) {
  .moviesListContainer {
    grid-template-columns: repeat(3, 1fr);
  }
}