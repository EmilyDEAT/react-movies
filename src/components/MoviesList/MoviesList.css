.moviesListContainer {
  width: 90%;
  margin: 5vh auto 5vh auto;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-template-rows: auto;
  grid-gap: 3vh;
}

.noFavorite {
  grid-column-start: 1;
  grid-column-end: 5;
}

@media screen and (min-width: 450px) and (max-width: 770px) {
  .moviesListContainer {
    grid-template-columns: repeat(4, 1fr);
  }
}

@media screen and (max-width: 450px) {
  .moviesListContainer {
    grid-template-columns: repeat(3, 1fr);
  }
}