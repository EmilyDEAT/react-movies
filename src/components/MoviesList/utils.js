// Get Movie Genre
const getGenre = (s) => {
  let idGenre = 0;
  if (s === "action") {
    idGenre = 28;
  } else if (s === "aventure") {
    idGenre = 12;
  } else if (s === "comedie") {
    idGenre = 35;
  } else if (s === "drame") {
    idGenre = 18;
  } else if (s === "familial") {
    idGenre = 10751;
  } else if (s === "fantastique") {
    idGenre = 14;
  }
  return idGenre;
};

export default getGenre;
