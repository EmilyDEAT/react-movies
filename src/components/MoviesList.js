import React, { useState, useEffect } from "react";
import axios from "axios";

import Movie from "./Movie";

import "./MoviesList.css";

const APIKey = "92b418e837b833be308bbfb1fb2aca1e";

const MoviesList = () => {
  const [moviesList, setMoviesList] = useState([]);

  const getMoviesList = async () => {
    const result = await axios.get(
      `https://api.themoviedb.org/3/movie/popular?api_key=${APIKey}&language=fr&page=1`
    );
    setMoviesList(result.data.results);
  };

  useEffect(getMoviesList, []);

  return (
    <div className="moviesListContainer">
      {moviesList.map((movie) => (
        <Movie movie={movie} key={movie.id} />
      ))}
    </div>
  );
};

export default MoviesList;
