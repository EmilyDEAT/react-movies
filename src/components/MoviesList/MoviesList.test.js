import React from "react";
import renderer from "react-test-renderer";

import MoviesList from "./MoviesList";
import getGenre from "./utils";

it("MoviesList renders correctly", () => {
  const props = { match: { params: { genre: "" } } };
  const moviesList = renderer.create(<MoviesList {...props} />).toJSON();
  expect(moviesList).toMatchSnapshot();
});

it("getGenre returns the right number", () => {
  expect(getGenre("action")).toEqual(28);
  expect(getGenre("aventure")).toEqual(12);
  expect(getGenre("comedie")).toEqual(35);
  expect(getGenre("drame")).toEqual(18);
  expect(getGenre("familial")).toEqual(10751);
  expect(getGenre("fantastique")).toEqual(14);
  expect(getGenre("")).toEqual(0);
});
