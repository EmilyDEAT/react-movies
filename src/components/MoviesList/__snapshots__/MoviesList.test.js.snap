// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList renders correctly 1`] = `
<div
  className="loader"
/>
`;
